crates/ra_hir/src/nameres/raw.rs

```diff
@@ -117,8 +118,14 @@ impl Index<Macro> for RawItems {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub(super) struct RawItem {
+    pub(super) attrs: Arc<[Attr]>,
```

Comment:
```suggestion
    pub(super) attrs: Option<Arc<[Attr]>>,
```

I think it might be worth it to micro-optimize the `Attr`s here, to avoid allocation for an empty slice